src/main/java/filter/expression/Qualifier.java

```diff
@@ -659,19 +659,7 @@ public static boolean hasUpdatedQualifier(FilterExpression expr) {
                 } else if (!bMilestone.isPresent()) {
                     return -1;
                 } else {
-                    Optional<LocalDate> aDueDate = aMilestone.get().getDueDate();
-                    Optional<LocalDate> bDueDate = bMilestone.get().getDueDate();
-
-                    if (!aDueDate.isPresent() && !bDueDate.isPresent()) {
-                        return 0;
-                    } else if (!aDueDate.isPresent()) {
-                        return 1;
-                    } else if (!bDueDate.isPresent()) {
-                        return -1;
-                    } else {
-                        return -(TurboMilestone.getDueDateComparator()
-                            .compare(aMilestone.get(), bMilestone.get()));
-                    }
+                    return -(TurboMilestone.getDueDateComparator().compare(aMilestone.get(), bMilestone.get()));
```

Comment:
This changes the behaviour of sorting. Is there any test case covering this?

From the old code, it seems to be: milestones no due date > milestone with due date sorted. Now we separate milestones no due date of different status.